marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/IOEndpointImpl.java

```diff
@@ -137,20 +133,24 @@ public void checkAllowedArgs(InputStream endpointState, SessionState session, In
         }
 
         private WorkPhase phase = WorkPhase.INITIALIZING;
-
-        private IOEndpointImpl endpoint;
-        private byte[]         endpointState;
-        private byte[]         workUnit;
         private SessionState   session;
         private CallContext callContext;
-
         private long           callCount = 0;
 
+        private CallerThreadPoolExecutor callerThreadPoolExecutor;
+        private LinkedBlockingQueue<CallContext> callContexts;
+
         BulkIOEndpointCallerImpl(IOEndpointImpl endpoint, CallContext callContext) {
             if (endpoint == null)
                 throw new IllegalArgumentException("null endpoint definition");
```

Comment:
Why is the BulkIOEndpointCallerImpl() constructor taking an endpoint if it's not doing anything with it?